### ISSUE TITLE
$foo > $int should infer that $foo is not null.

### ISSUE THREAD

**Issue by VincentLanglet** [vimeo/psalm#6274]
```
var_dump(null >= -1);
```
is returning false, so I assume that `null > $int` will return true.

So in the following code https://psalm.dev/r/dc513a3015, I would expect the `PossiblyNullOperand` to not be reported.

**Comment by weirdan** [comment]
https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.types

Frankly, I don't think this needs to be allowed.

**Comment by VincentLanglet** [comment]
> > I assume that null > $int will return true.
> 
> Why? `null` is less than any integer (except 0): https://3v4l.org/c4j4v

Yes, I meant `null > $int` will always return false. Sorry, I edited.

> Frankly, I don't think this needs to be allowed.

I was expecting such a comment.

The best could be to
- Infer that the $int cannot be null (because it's true).
- Add a new error `PossiblyNullComparison`, the same way there is a `PossiblyNullOperand`.

**Comment by weirdan** [comment]
The latter sounds sensible.
